packages/bot-web-ui/src/components/journal.jsx

```diff
@@ -93,11 +93,102 @@ const Tools = ({ checked_filters, filters, filterMessage }) => (
                 );
             })}
         </div>
-        {/* <div className='tools__container-download'>
-            <Icon icon='IcDownload' />
-        </div> */}
-    </div>
-);
+    );
+};
+
+const FilterDialog = ({
+    toggle_ref,
+    checked_filters,
+    filters,
+    filterMessage,
+    is_filter_dialog_visible,
+    toggleFilterDialog,
+}) => {
+    const wrapper_ref = React.useRef();
+
+    React.useEffect(() => {
+        const clickOutsideListener = event => {
```

Comment:
We have an upcoming shared hook for this. Could you add a todo that explains we should use the shared hook in the future?

Reply:
@aaron-binary @nijil-binary it's merged now :) but will need to be exported and imported from packages/components

Reply:
TODO added.